### PR TITLE
Remove duplicated breadcrumbs when syncing with iOS/macOS

### LINKED ISSUE #1270: Duplicated `BreadCrumbs` from sentry_logging
_Platform:_
- [ ]  Dart
- [x]  Flutter Android or iOS
- [ ]  Flutter Web

_IDE:_
- [x]  VSCode
- [ ]  IntelliJ/AS
- [ ]  XCode
- [ ]  Other, which one?

_split-debug-info and obfuscate (Flutter Android or iOS) or CanvasKit (Flutter Web):_
- [ ]  Enabled
- [x]  Disabled

_Platform installed with:_
- [x] pub.dev
- [ ] GitHub

Output of the command `flutter doctor -v` below:   
```
[✓] Flutter (Channel stable, 3.7.1, on macOS 12.6.2 21G320 darwin-x64, locale en-US)
    • Flutter version 3.7.1 on channel stable at /Users/svetozar/soft/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (6 days ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1
    • Pub download mirror https://pub.dev/

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc1)
    • Android SDK at /Users/svetozar/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[!] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.75.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • iPhone (Светозар) (mobile) • 00008120-00023C1C1420C01E • ios        • iOS 16.3 20D47
    • macOS (desktop)            • macos                     • darwin-x64 • macOS 12.6.2 21G320 darwin-x64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

The version of the SDK (See pubspec.lock):
**6.19.0**

---
I have the following issue:

I added a logging integration to my project. But **almost** all logs are duplicated in sentry.
Tested only on iOS for now.

**Steps to reproduce:**
1. Add logging integration
2. Log some messages
3. Throw an exception

**Actual result:**

Logs can be duplicated in two ways (I am using LogRecord.sequenceNumber instead of full log):

- immidiately one after another: 1 -> 1 -> 2 -> 2...
- messed up: 11 -> 12 -> 13 -> 12 -> 13 -> 11...

Screenshot illustrates the first case:
Screen Shot 2023-02-07 at 13 17 21

**Expected result:**
- Logs are not duplicated

P.S. looks similar to #946

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

By fixing the issue that scope sync was never enabled on iOS with #1196, we introduced the issue that breadcrumbs could be duplicated because of bi-directional sync by calling `addBreadcrumbs`, and loading the native context with `loadContexts` to enrich events, which also includes the native breadcrumbs.

- Create breadcrumbs with correct timestamp when syncing down to native layer through `addBreadcrumbs`
- Remove duplicate breadcrumbs when syncing up from native layer through `loadContexts`

## :bulb: Motivation and Context

Closes #1270 

## :green_heart: How did you test it?

Added unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

This approach in this PR fixes the issue, but we now have to compare all breadcrumbs with each other, which could get slow with a larger number of breadcrumbs.

Another way would be to just remove syncing breadcrumbs down to the native layer.

### CODE REVIEW

**Review by brustolin — APPROVED** [review]
LGTM

File: flutter/lib/src/integrations/load_contexts_integration.dart [hunk 1]
```diff
@@ -139,13 +140,27 @@ class _LoadContextsIntegrationEventProcessor extends EventProcessor {
...
+            if (mapEquals(newBreadcrumbJson, breadcrumbJson)) {
+              containsDuplicate = true;
+              break;
+            }
+          }
+
+          if (!containsDuplicate) {
+            final newBreadcrumb = Breadcrumb.fromJson(newBreadcrumbJson);
+            breadcrumbs.add(newBreadcrumb);
+          }
```
Comment (marandaneto) [inline flutter/lib/src/integrations/load_contexts_integration.dart]: IIRC @krystofwoldrich and I talked about this already.
If `scopeSync` is enabled, use the crumbs from the bridge so you don't need to duplicate anything (it contains all the crumbs in the right order already).
If `scopeSync` is disabled, use the crumbs from the Dart scope.

File: flutter/ios/Classes/SentryFlutterPluginApple.swift [hunk 2]
```diff
@@ -53,6 +53,14 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
 
     }
 
+    private lazy var dateFormatter = {
+        let formatter = DateFormatter()
+        formatter.locale = Locale(identifier: "en_US_POSIX")
+        formatter.timeZone = TimeZone(abbreviation: "UTC")
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSSSSXXX"
+        return formatter
+    }()
```
Comment (marandaneto) [inline flutter/ios/Classes/SentryFlutterPluginApple.swift]: Would be possible to use https://github.com/getsentry/sentry-cocoa/blob/main/Sources/Sentry/NSDate%2BSentryExtras.m directly?
`getIso8601FormatterWithMillisecondPrecision` function should do.
The `dateFormat` should be `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'` to match the Cocoa SDK

File: flutter/lib/src/integrations/load_contexts_integration.dart (hunk 1 above)
Comment (marandaneto) [reply flutter/lib/src/integrations/load_contexts_integration.dart]: https://github.com/getsentry/sentry-react-native/blob/1542a6003154377ad611e0f0eca8902d3962fdef/src/js/integrations/devicecontext.ts#L92-L97

the RN SDK does not have a `enableScopeSync`, but we do.

**Review by marandaneto — APPROVED** [review]
Thanks @denrase

File: flutter/test/integrations/init_native_sdk_integration_test.dart [hunk 3]
```diff
@@ -53,7 +53,7 @@ void main() {
         'maxCacheItems': 30,
         'sendDefaultPii': false,
         'enableOutOfMemoryTracking': true,
-        'enableNdkScopeSync': false,
+        'enableNdkScopeSync': true,
```
Comment (marandaneto) [inline flutter/test/integrations/init_native_sdk_integration_test.dart]: @denrase is this intentional?

**Pull request merged**